### ISSUE TITLE
using HTTPResponseCompressor results in invalid encoding headers

### ISSUE THREAD

**Issue by ilyapuchka** [apple/swift-nio#717]
### Expected behavior
HTTPResponseCompressor sets encoding headers to the expected value, i.e. if a response created by the previous handler in the pipeline already set `Content-Encoding` header, the compressor should respect this.

### Actual behavior
When the previous handler in the pipeline sets `Content-Encoding` header to i.e. `gzip` value, then HTTPResponseCompressor, if it detects that response should be compressed with `gzip` (based on `Accept-Encoding` header), adds the same value in the header, which results in its value being `gzip, gzip`, which is then not properly handled by the client (decompression fails). Instead, the value should be just `gzip`.

### Steps to reproduce

1. In `NIOHTTP1Server/main.swift` add `HTTPResponseCompressor` handler after `HTTPHandler`
2. Make `HTTPHandler` set `Content-Encoding` and `Accept-Encoding` headers to `gzip`.

or

1. In [SwiftNIOMock](https://github.com/ilyapuchka/SwiftNIOMock) comment out [SwiftNIOMock/Server.swift:267](https://github.com/ilyapuchka/SwiftNIOMock/blob/58bed932e93da7fad7a1aaf644587f60574d48a9/SwiftNIOMock/Sources/SwiftNIOMock/Server.swift#L267)

2. Run tests, the assert in `testCanRedirectRequestAndInterceptResponse` will fail because no data is received (because it is failing to be decoded)

### If possible, minimal yet complete reproducer code (or URL to code)

Problematic code:

https://github.com/apple/swift-nio/blob/305ee9818b8d6de85433bd4da7c70aa860cd0189/Sources/NIOHTTP1/HTTPResponseCompressor.swift#L123

### SwiftNIO version/commit hash

1.12.0

### Swift & OS version (output of `swift --version && uname -a`)

```
Target: x86_64-apple-darwin17.7.0
Darwin b-0800.local 17.7.0 Darwin Kernel Version 17.7.0: Thu Jun 21 22:53:14 PDT 2018; root:xnu-4570.71.2~1/RELEASE_X86_64 x86_64
```

**Comment by Lukasa** [comment]
Yup, good spot. We should fix this. For now I'm scheduling for 2.0 just because it seems like the likely timescale, but if someone writes a fix we'd consider back porting to 1.12.

**Comment by ilyapuchka** [comment]
Do you think just using the `removeOrAdd` when adding this header be enough as a solution?

**Comment by weissi** [comment]
this is fixed by #731
